ReSharper.FSharp/src/FSharp.TypeProviders.Protocol/src/Utils/Extensions.cs

```diff
@@ -1,16 +1,21 @@
 ﻿using System;
 using System.Collections.Generic;
-using FSharp.Compiler;
 using JetBrains.Util.dataStructures;
+using static FSharp.Compiler.ExtensionTyping;
 
 namespace JetBrains.ReSharper.Plugins.FSharp.TypeProviders.Protocol.Utils
 {
   public static class Extensions
   {
-    public static string GetLogName(this ExtensionTyping.ProvidedAssembly assembly) =>
+    public static string GetLogName(this ProvidedAssembly assembly) =>
       assembly.GetName().Version == null ? "generated assembly" : assembly.FullName;
 
-    public static void RemoveAll<T, TU>(this Dictionary<T, TU> dict, Func<KeyValuePair<T, TU>, bool> match)
+
+    public static bool IsCreatedByProvider(this ProvidedType providedType) =>
+      //FSharp.TypeProviders.SDK/ProvidedTypes.fsi
+      providedType.RawSystemType.GetType().Name == "ProvidedTypeDefinition";
```

Comment:
Does this check work properly for generic metadata types with provided type substitutions, e.g. `IList<SomeProvidedType>`?

Reply:
yes, and it's checked [here](https://github.com/JetBrains/resharper-fsharp/blob/c671e1de5f1f4e13cb70728a9df2215b5437396e/rider-fsharp/testData/typeProviders/TypeProvidersCacheTest/typing/gold/typing.gold#L203)

Reply:
Hm, from the test data it looks like it got into metadata/shared types group. Is it actually expected that provided type instantiation is shared between type providers? How is it different from a case where a provided type is not used in a substitution and is not allowed to be shared? Or am I reading the test dump wrong?

Reply:
Fair remark. We should cache them separately as well.